connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/a2a/client/configuration/A2aConnectorConfiguration.java

```diff
@@ -0,0 +1,106 @@
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.agenticai.a2a.client.configuration;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.connector.agenticai.a2a.client.A2aConnectorFunction;
+import io.camunda.connector.agenticai.a2a.client.api.A2aAgentCardFetcher;
+import io.camunda.connector.agenticai.a2a.client.api.A2aMessageSender;
+import io.camunda.connector.agenticai.a2a.client.api.A2aRequestHandler;
+import io.camunda.connector.agenticai.a2a.client.api.A2aSdkClientFactory;
+import io.camunda.connector.agenticai.a2a.client.api.A2aSendMessageResponseHandler;
+import io.camunda.connector.agenticai.a2a.client.api.TaskPoller;
+import io.camunda.connector.agenticai.a2a.client.convert.DocumentToPartConverter;
+import io.camunda.connector.agenticai.a2a.client.convert.DocumentToPartConverterImpl;
+import io.camunda.connector.agenticai.a2a.client.convert.PartsToContentConverter;
+import io.camunda.connector.agenticai.a2a.client.convert.PartsToContentConverterImpl;
+import io.camunda.connector.agenticai.a2a.client.impl.A2aAgentCardFetcherImpl;
+import io.camunda.connector.agenticai.a2a.client.impl.A2aMessageSenderImpl;
+import io.camunda.connector.agenticai.a2a.client.impl.A2aRequestHandlerImpl;
+import io.camunda.connector.agenticai.a2a.client.impl.A2aSendMessageResponseHandlerImpl;
+import io.camunda.connector.agenticai.a2a.client.impl.TaskPollerImpl;
+import io.camunda.connector.agenticai.a2a.client.sdk.A2aSdkClientFactoryImpl;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBooleanProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConditionalOnBooleanProperty(
+    value = "camunda.connector.agenticai.a2a.client.enabled",
+    matchIfMissing = true)
+public class A2aConnectorConfiguration {
+
+  @Bean
+  @ConditionalOnMissingBean
+  public DocumentToPartConverter documentToPartConverter(ObjectMapper objectMapper) {
+    return new DocumentToPartConverterImpl(objectMapper);
+  }
+
+  @Bean
+  @ConditionalOnMissingBean
+  public PartsToContentConverter partsToContentConverter() {
+    return new PartsToContentConverterImpl();
+  }
+
+  @Bean
+  @ConditionalOnMissingBean
+  public A2aSendMessageResponseHandler a2aSendMessageResponseHandler(
+      PartsToContentConverter partsToContentConverter) {
+    return new A2aSendMessageResponseHandlerImpl(partsToContentConverter);
+  }
+
+  @Bean
+  @ConditionalOnMissingBean
+  public ScheduledExecutorService taskPollerExecutor() {
+    return Executors.newScheduledThreadPool(2);
```

Comment:
Should we make the value configurable by introducing an a2a specific config property structure with 2 as default value?